extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java

```diff
@@ -996,19 +996,35 @@ private ResultHandle createMultipartForm(MethodCreator methodCreator, ResultHand
             if (Modifier.isStatic(field.flags())) {
                 continue;
             }
-            if (!Modifier.isPublic(field.flags())) {
-                throw new IllegalArgumentException("Non-public field found in a multipart form data class "
+
+            String fieldName = field.name();
+            ResultHandle fieldValue = null;
+            if (Modifier.isPublic(field.flags())) {
+                fieldValue = methodCreator.readInstanceField(field, methodParam);
+
+            } else {
+                String getterName = "get" + fieldName.substring(0, 1).toUpperCase() + fieldName.substring(1);
```

Comment:
I'm wondering if when we have a getter and a public field we shouldn't use the getter

Reply:
That's a good point. I'll update